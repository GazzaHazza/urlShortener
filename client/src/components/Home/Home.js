import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { selectSuggestedPlace } from "./../../reducers/url";
class Home extends Component {
  render() {
    return <div className="home" />;
  }
}
// function mapStateToProps(state, props) {
// 	return {
//         search: state.search
//     };
// }
// function mapDispatchToProps(dispatch) {
// 	return {
// 		selectSuggestedPlace: bindActionCreators(selectSuggestedPlace, dispatch)
// 	}
// }
// export default connect(mapStateToProps, mapDispatchToProps)(Home);
export default Home;
