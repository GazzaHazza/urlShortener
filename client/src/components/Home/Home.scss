@import './../../styles/_colors.scss';

.home {
    &__container {
        padding: 20px;
        background: $color1;
    }
}