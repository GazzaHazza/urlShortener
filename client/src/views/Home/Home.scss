.home {
    &__container {
        width:75%;
        margin: 0 auto; 
        @media all and (max-width: 520px) {
            width:90%;
        }
    }
}