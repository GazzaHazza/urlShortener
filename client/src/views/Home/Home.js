import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import FadeIn from "react-fade-in";
import { sendNewUrl } from "./../../reducers/url";
import UrlForm from "../../components/UrlForm/UrlForm";
import Message from "../../components/Message/Message";
import "./Home.css";

class Home extends Component {
  generateNewUrl = url => {
    this.props.sendNewUrl && this.props.sendNewUrl(url);
  };
  render() {
    const {
      url: { message, hasError, hasAdded, shortUrl, orginalUrl, isBusy }
    } = this.props;
    const messageComp =
      message && (hasAdded || hasError)
        ? <FadeIn>
            <Message
              message={message}
              hasError={hasError}
              hasAdded={hasAdded}
              shortUrl={shortUrl}
              orginalUrl={orginalUrl}
            />
          </FadeIn>
        : null;
    return (
      <div className="home">
        <div className="home__container">
          <h1 className="home__title">Url shortener....</h1>
          <h2 className="home__subtitle">
            Enter url below and use "short" version.
          </h2>
          <UrlForm onClickGenerate={this.generateNewUrl} isBusy={isBusy} />
          {messageComp}
        </div>
      </div>
    );
  }
}
Home.propTypes = {
  url: PropTypes.shape({
    hasError: PropTypes.bool,
    hasAdded: PropTypes.bool,
    isBusy: PropTypes.bool,
    shortUrl: PropTypes.string,
    orginalUrl: PropTypes.string,
    message: PropTypes.string
  })
};
export default connect(
  state => {
    return {
      url: state.url
    };
  },
  {
    sendNewUrl
  }
)(Home);
