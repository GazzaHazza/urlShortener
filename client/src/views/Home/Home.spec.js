import React from "react";
import { shallow } from "enzyme";
import { Home } from "./Home";
it("renders correctly with default props", () => {
  const mockProps = {
    url: {
      isBusy: false,
      hasAdded: false,
      hasFetched: false,
      hasError: false,
      message: null,
      shortUrl: null,
      originalUrl: null
    }
  };
  const wrapper = shallow(<Home {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly with isBusy prop set true", () => {
  const mockProps = {
    url: {
      isBusy: true,
      hasAdded: false,
      hasFetched: false,
      hasError: false,
      message: null,
      shortUrl: null,
      originalUrl: null
    }
  };
  const wrapper = shallow(<Home {...mockProps} isBusy />);
  expect(wrapper).toMatchSnapshot();
});

it("renders correctly with hasError prop set true", () => {
  const mockProps = {
    url: {
      isBusy: false,
      hasAdded: false,
      hasFetched: false,
      hasError: true,
      message: "test",
      shortUrl: "test",
      originalUrl: null
    }
  };
  const wrapper = shallow(<Home {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("should call sendNewUrl on when generalNewUrl is called", () => {
  const mockProps = {
    url: {
      isBusy: false,
      hasAdded: false,
      hasFetched: false,
      hasError: true,
      message: "test",
      shortUrl: "test",
      originalUrl: null
    },
    sendNewUrl: jest.fn()
  };
  const wrapper = shallow(<Home {...mockProps} />);
  wrapper.instance().generateNewUrl("hello");
  expect(mockProps.sendNewUrl).toBeCalledWith("hello");
});
