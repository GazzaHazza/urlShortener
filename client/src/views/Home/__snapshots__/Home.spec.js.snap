// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default props 1`] = `
<styled.div>
  <styled.h1>
    Shorten your link....
  </styled.h1>
  <styled.h2>
    Enter a url below, click generate and then use the "short" version.
  </styled.h2>
  <UrlForm
    isBusy={false}
    onClickGenerate={[Function]}
  />
</styled.div>
`;

exports[`renders correctly with hasError prop set true 1`] = `
<styled.div>
  <styled.h1>
    Shorten your link....
  </styled.h1>
  <styled.h2>
    Enter a url below, click generate and then use the "short" version.
  </styled.h2>
  <UrlForm
    isBusy={false}
    onClickGenerate={[Function]}
  />
  <FadeIn>
    <MessageBox
      hasAdded={false}
      hasError={true}
      message="test"
      shortUrl="test"
    />
  </FadeIn>
</styled.div>
`;

exports[`renders correctly with isBusy prop set true 1`] = `
<styled.div>
  <styled.h1>
    Shorten your link....
  </styled.h1>
  <styled.h2>
    Enter a url below, click generate and then use the "short" version.
  </styled.h2>
  <UrlForm
    isBusy={true}
    onClickGenerate={[Function]}
  />
  <Loading />
</styled.div>
`;
